en/docs/concepts/identity-federation.md

```diff
@@ -0,0 +1,44 @@
+
+----
+
+## How does identity federation work?
+
+The main parties involved in an identity federation flow are:
+
+- **Identity Provider (IdP)**: The application in which the user has created his/her account is called the **identity provider (IdP)** and it is responsible for maintaining the user's digital identity. 
+
+- **Relying Party (RP)**: The application that is depending upon the identity provider to maintain and vouch for the user's identity is called the **relying party (RP)**. 
+
+The identity federation flow is as follows:
+
+1. The user attempts to access a service provided by the RP or log in to the RP application. 
+
+2. The RP needs to authenticate the user and is configured to trust the IdP to do this on behalf of it. 
+
+3. The unauthenticated user is redirected to the IdP's login page and prompted to provide their credentials to authenticate themselves to the IdP. 
+
+    In some cases, the RP may check session data with the IdP and skip this step if the user is already authenticated with the IdP. For more information, see [Single Sign-On vs Identity Federation](TODO:link-to-concept).
+
+4. If the user is successfully validated and authenticated, the user is redirected back to the RP application with an access token, which allows the user to access the application. 
+
+(TODO: insert-diagram)
+
+----
+
+## Identity federation with WSO2 Identity Server
+
+In a real-world implementation, organizations may have multiple relying party applications that need trust relationships with multiple identity providers, in order to authenticate users. Another requirement may be to enable relying party applications or services to provide access to identities that are *outside* the organization's traditional boundary/trust domain. These scenarios require an **identity broker** or a federated identity management system such as WSO2 Identity Server, which already has trust relationships with external identity providers and mechanisms. The identity broker can be used to set up a federation arrangement for multiple relying party applications. 
+
+WSO2 Identity Server has federated authenticators for social idenity providers such as Google, Facebook, LinkedIn, Twitter, Salesforce, etc., as well as authenticators for protocols such as SAML 2.0, OpenID Connect, OAuth 2.0, and WS-Federation (passive). A federated authenticator for a protocol (such as SAML) has to be associated with an identity provider using the same protocol. ,
+
+You can register relying party applications in WSO2 Identity Server and configure which federated authenticators are to be used for each relying party. As it is highly extensible, you can also develop your own federated authenticator and plug it into WSO2 IS if required. 
```

Comment:
Make this WSO2 Identity Server or introduce it as WSO2 IS somewhere in the beginning itself.